panda/src/navigation/navMeshQuery.cxx

```diff
@@ -0,0 +1,198 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file navMeshQuery.cxx
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+#include "navMeshQuery.h"
+#include "DetourNavMeshQuery.h"
+#include "lvecBase3.h"
+#include <iostream>
+
+NavMeshQuery::NavMeshQuery():
+  _nav_query(0) {
+  _nav_query = dtAllocNavMeshQuery();
+}
+
+NavMeshQuery::NavMeshQuery(PT(NavMesh) nav_mesh):
+  _nav_query(0) {
+  
+  _nav_query = dtAllocNavMeshQuery();
+  dtStatus status = _nav_query->init(nav_mesh->get_nav_mesh(), 2048);
+
+  if (dtStatusFailed(status)) {
+    std::cout<<"\nCannot set nav query!\n";
+  }
+
+}
+
+NavMeshQuery::~NavMeshQuery() {
+  dtFreeNavMeshQuery(_nav_query);
+}
+
+bool NavMeshQuery::set_nav_query(PT(NavMesh) nav_mesh) {
```

Comment:
These functions all need doc strings which will be used for our generated API reference.